app/controllers/tickets_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class TicketsController < ApplicationController
+  def show
+    respond_to do |format|
+      format.html do
+        @ticket_id = params.require(:id)
+        render :index
+      end
+      format.json do
+        ticket = Ticket.find(params.require(:id))
+        requester_stakeholders = ticket.user_stakeholders(current_user)
+
+        return head :unauthorized if requester_stakeholders.empty?
+        render json: {
+          ticket: ticket,
+          requester_stakeholders: requester_stakeholders,
+        }
+      end
+    end
+  end
+
+  def update_status
+    ticket = Ticket.find(params.require(:ticket_id))
+    ticket_status = params.require(:ticket_status)
+
+    return head :unauthorized unless ticket.action_allowed?(:update_status, current_user)
+
+    ActiveRecord::Base.transaction do
+      ticket.metadata.update!(status: ticket_status)
+      TicketLog.create!(
+        ticket_id: ticket.id,
+        log: "Ticket status changed to #{ticket_status} by #{current_user.name}.",
+      )
+    end
+    status_changed = ticket.metadata.update(status: ticket_status)
```

Comment:
You're executing `ticket.metadata.update` twice: Once with an `!` within the transaction, and then once immediately after the transaction.

But it's the same ticket and the same `ticket_status` variable being passed, so I wonder if this is a whoopsie?

Reply:
Also, I want to point out that the return value of `update` does not indicate whether the status _changed_. The returned boolean only indicates whether the write operation to the DB was successful or not. It may be that nothing changed, in which case a kind of "no-op" SQL update will be performed, and the method will still return `true`.

The most notable use-case for this return value is validations: If you pass arguments to `update` that would make your model invalid (as defined by your own validations on the AR model), then the return value will be `false`.

Reply:
Oh sorry this was a whoopsie. I've removed one. Ya I knew the returned boolean just represents whether it's success or not, I shouldn't have named it `status_changed`. Anyway I have hard-coded `render json: { success: true }` now.